GCI "Filter layers that are exclusive to selected frameworks"

ZeroZeroJedenJeden opened the pull request:
Added filters to show/hide layers in layers bar to selected framework.

thatbrguy commented:
Hey @ZeroZeroJedenJeden few suggestions before I do a full review. 
- Can you update the instructions in this [documentation](https://github.com/Cloud-CV/Fabrik/blob/master/tutorials/adding_new_layers.md) to add their layer name to your variables?
- Please share a gif or video demonstrating your feature in the Fabrik Gitter [channel](https://gitter.im/Cloud-CV/Fabrik).

thatbrguy commented:
@ZeroZeroJedenJeden That's some excellent work, well done! I have a few comments based on my observations:
- While using the Keras filter, layers like "Input" and "Convolution" get filtered out. However, those layers are available in Keras. Please check your filter arrays if they are missing/having any irrelevant elements.
- It would great if you could have the checkmark like: `Keras [X]` instead of `[X] Keras`. 
- Currently I have to click the checkmark area to toggle the filter. Can you change it such that clicking anywhere on the `Keras [X]` dropdown would toggle the filter (Instead of clicking on the checkmark `[X]`)?
- Your functionality works well in conjunction with the search bar. So that's great!

@Ram81 @yashdusing It'll be great if you guys can review this functionality as well.

ZeroZeroJedenJeden commented:
- "Input" and "Convolution" added to filter array.

- ```[X] Keras``` changed to ```Keras [X]```

- checkmark work on all button area.

- removed all comented code and removed all ```error  Mixed spaces and tabs  no-mixed-spaces-and-tabs``` errors

@thatbrguy If you can, check the work.
>

yashdusing commented:
The functionality's great. But I think :
- You should add some margin to filter layer's search bar in the bottom and the top. This looks a little clustered.
- Try to align the button along with search bar. The search bar takes up both the Insert layer heading's area as well as search bar's. Try to bring it to search bar's level and keep it within that bounds.
Apart from this : LGTM

ZeroZeroJedenJeden commented:
@yashdusing I see it, but to do it, I'll modify search bar placing system. Search bar is not my GCI task, so I wait what @thatbrguy will say.

thatbrguy commented:
@ZeroZeroJedenJeden Hey, good progress! However there are still some layer discrepancies. For instance, `Crop` is Caffe only, but is available on Keras and TensorFlow as well. Similarly, `LRN` is available in `Keras`. `Input` is available on TensorFlow as well and so on.

Please go through `data.js` and the files in `tensorflow_app`, `keras_app` and `caffe_app` and verify if everything has been accounted for.

ZeroZeroJedenJeden commented:
My brain has stopped working. I got elements for filter from #409 PR without verification, where Input missing, LRN only for cafee and Crop for all frameworks. I going to rewrite all arrays with official domumentacions.

ZeroZeroJedenJeden commented:
- Updated filters from ```data.js```

ZeroZeroJedenJeden commented:
- Modifed placing system.

thatbrguy commented:
@ZeroZeroJedenJeden Glad that you're actively verifying the layers, good work! @yashdusing Could you verify if the filter functionality has all the proper layers for each framework? I may have missed something. 

I got a few points as well:
- Looks like `content.js`, `pane.js` and `filterbar.js` got added twice in your commits. Please remove the redundancies.
- Placement looks weird all of a sudden, did you change anything? You can modify it to the older version if you cant fix it. I've attached an image showing the problem below:
![filter](https://user-images.githubusercontent.com/30566404/47955318-594f1200-dfbc-11e8-8cb3-6b5a6ae59409.png)

yashdusing commented:
I have the same display issue as @thatbrguy so please fix that bug.

ZeroZeroJedenJeden commented:
I pushed ```content.js```, ```pane.js``` and ```filterbar.js``` to master folder. This create this look bug and double files. It should be good now.

thatbrguy commented:
@ZeroZeroJedenJeden Hello. Yes, placement looks perfect now. 

About the layers, @yashdusing and I will verify if everything has been properly added and let you know if something's missing.

yashdusing commented:
@ZeroZeroJedenJeden Please look into the comments I've made and fix those. Apart from those comments : LGTM

thatbrguy commented:
@ZeroZeroJedenJeden Looks good to me, excellent work :+1:

Ram81 commented:
@ZeroZeroJedenJeden style gets messed up on chrome, can you fix this.
<img width="1440" alt="screen shot 2018-11-05 at 2 24 57 pm" src="https://user-images.githubusercontent.com/16323427/47987877-ab9c4a00-e106-11e8-8939-617130a972cf.png">

Ram81 commented on ide/static/js/content.js:
```diff
@@ -1308,17 +1309,20 @@ class Content extends React.Component {
              <h5 className="sidebar-heading">LOGIN</h5>
              <Login setUserId={this.setUserId} setUserName={this.setUserName}></Login>
+             <h5 className="sidebar-heading">
+             <div id="insert-layer-sign">INSERT LAYER</div>
```
Correct the Indentation

Ram81 commented on ide/static/js/content.js:
```diff
@@ -1308,17 +1309,20 @@ class Content extends React.Component {
              />
              <div className="text-center">
-              <Tabs selectedPhase={this.state.selectedPhase} changeNetPhase={this.changeNetPhase} />
+             <Tabs selectedPhase={this.state.selectedPhase} changeNetPhase={this.changeNetPhase} />
```
Fix the indentation

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+      "Deconvolution_Button", "Regularization_Button", "Softsign_Button", "Upsample_Button", "Pooling_Button",
+      "LocallyConnected_Button", "Input_Button", "Convolution_Button", "LRN_Button", "DepthwiseConv_Button", "Flatten_Button",
+      "Reshape_Button", "Concat_Button", "Softplus_Button", "HardSigmoid_Button", "Dropout_Button", "Eltwise_Button"];
+      var TensorFlowLayers = ["Input_Button", "Convolution_Button", "Pooling_Button", "DepthwiseConv_Button", "InnerProduct_Button",
```
Add a new line above this

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+      "LRN_Button", "Concat_Button", "Flatten_Button", "BatchNorm_Button", "Deconvolution_Button", "Sigmoid_Button", 
+      "Softplus_Button", "Softsign_Button", "ELU_Button", "ReLU_Button", "Softmax_Button", "TanH_Button", "SELU_Button",
+      "Dropout_Button", "Eltwise_Button"];
+      var CaffeLayers = ["ImageData_Button", "HDF5Data_Button", "HDF5Output_Button", "Input_Button", "WindowData_Button",
```
Add a newline above this

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
```
Rename the file to `filterBar.js`

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+      var CheckBoxC = document.getElementById("CheckBoxC");
+      var visible = [];
+
+    let CheckBox = document.getElementById(cbid);
```
Follow the camel case style for variable names

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+         visible = visible.concat(CaffeLayers);
+      }
+      
+            for (let elem of $('.drowpdown-button')) {
```
Uneven Indentation

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+                  </ul>
+                </div>
+              </div>
+              </div>
```
Again Indentation fixes

Ram81 commented on ide/static/css/searchbar_style.css:
```diff
@@ -16,8 +17,8 @@
 	position: absolute;
-	top: 36px;
-	left: 20px;
+	top: 2px;
```
Rather than hardcoding the values you can assign relative values.

yashdusing replied:
@ZeroZeroJedenJeden try using relative values to see if that solves the layout bug

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+                if(id == visible[j]){
+                    elem.classList.remove("hide");
+                    j = visible.length + 1;
+                }else{
```
space before `else`

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+            for (let elem of $('.drowpdown-button')) {
+                for (let j = 0; j < visible.length; j++){
+                let id = elem.id;
+                if(id == visible[j]){
```
space after `)` and before `{`

Ram81 commented on ide/static/js/filterbar.js:
```diff
@@ -0,0 +1,103 @@
+                    <li>
+                        <a className="btn" onClick={this.changeEvent.bind(this, "CheckBoxA")}>
+                        <label className="filter">Keras</label>
+                        <input type="checkbox" id="CheckBoxA" value="A" onClick={this.changeEvent.bind(this, "CheckBoxA")} />
```
Rename checkbox ids to be more specific for functionality

ZeroZeroJedenJeden commented:
@Ram81 About this "style bug" on chrome. On my chrome all looks good. Try clean chrome cache, like ```ctrl + R``` and ```ctrl + F5``` or clean site cache in chrome settings.

Ram81 commented:
@ZeroZeroJedenJeden the size of filter button still looks to big. Is it same for you?

ZeroZeroJedenJeden commented:
@Ram81 The size of filter button is the same size as topBar buttons (import, export etc).

yashdusing commented:
Hello @ZeroZeroJedenJeden ! It seems that this task is still not yet done and you have simultaneously taken up another task for fixing layer chopiness. We would like you to finish this task up first and have the PR approved before taking up the other one as it violates the rule of taking up more than one tasks at a time.

yashdusing commented:
@utsavgarg

ZeroZeroJedenJeden commented:
@yashdusing Right, but after @thatbrguy "LGTM"comment I could take next task on GCI site. I took next task and after this @Ram81 wrote comments to this task.

yashdusing commented:
@ZeroZeroJedenJeden sorry for the confusion. I thought that the PR wasn't approved so the task might not have the approval. You can continue your work on the layer chopiness. Ask for help if you're stuck

ZeroZeroJedenJeden commented:
To @Ram81 big button comment. Are this look better?
![image](https://user-images.githubusercontent.com/44526720/48083119-a5849700-e1f4-11e8-9711-454b83e110f7.png)
